packages/gatsby/src/utils/page-data.js

```diff
@@ -36,17 +45,69 @@ const write = async ({ publicDir }, page, result) => {
   store.dispatch({
     type: `ADD_PAGE_DATA_STATS`,
     payload: {
-      filePath,
+      filePath: outputFilePath,
       size: pageDataSize,
     },
   })
 
-  await fs.outputFile(filePath, bodyStr)
+  await fs.outputFile(outputFilePath, bodyStr)
+  return body
+}
+
+const flush = async () => {
+  const { pendingPageDataWrites, components, pages, program } = store.getState()
+
+  const { pagePaths, templatePaths } = pendingPageDataWrites
+
+  const pagesToWrite = Array.from(templatePaths).reduce(
+    (set, componentPath) => {
+      const { pages } = components.get(componentPath)
```

Comment:
This code is never hit in this PR (because of https://github.com/gatsbyjs/gatsby/pull/24808/files#r436020425 ), but we should do similar `null`/`undefined` check  as we do for pending pages in https://github.com/gatsbyjs/gatsby/pull/24808/files#diff-89886096f2d57a6409f15d26edcbc6c4R74-R80 before reaching to `pages` (`components.get(componentPath)` might not find anything)

Reply:
Indeed, added